ksmt-yices/src/main/kotlin/org/ksmt/solver/yices/KYicesInternalizationContext.kt

```diff
@@ -0,0 +1,90 @@
+package org.ksmt.solver.yices
+
+import com.sri.yices.Yices
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KExpr
+import org.ksmt.sort.KSort
+import org.ksmt.utils.NativeLibraryLoader
+
+open class KYicesInternalizationContext {
+    protected val expressions = HashMap<KExpr<*>, YicesTerm>()
+    private val yicesExpressions = HashMap<YicesTerm, KExpr<*>>()
+    protected val sorts = HashMap<KSort, YicesSort>()
+    private val yicesSorts = HashMap<YicesSort, KSort>()
+    protected val decls = HashMap<KDecl<*>, YicesDecl>()
+    private val yicesDecls = HashMap<YicesDecl, KDecl<*>>()
+
+    fun removeDecl(decl: KDecl<*>) {
```

Comment:
It is better to clarify the following:
1. If we internalize a decl during constant internalization, we won't internalize the same decl again, since we have an internalized constant. 
2.  If we internalize a decl during variable internalization, we don't need decl since we have a variable expression.

Also, it's not clear, why internalized variable will not overwrite previously internalized constant?